src/Security/MemberAuthenticator/SessionAuthenticationHandler.php

```diff
@@ -0,0 +1,105 @@
+<?php
+
+namespace SilverStripe\Security\MemberAuthenticator;
+
+use SilverStripe\Control\Cookie;
+use SilverStripe\Security\Member;
+use SilverStripe\Control\HTTPRequest;
+use SilverStripe\Control\Session;
+use SilverStripe\Control\Director;
+use SilverStripe\Security\AuthenticationHandler;
+use SilverStripe\Security\IdentityStore;
+
+/**
+ * Authenticate a member pased on a session cookie
+ */
+class SessionAuthenticationHandler implements AuthenticationHandler, IdentityStore
+{
+
+    private $sessionVariable;
+
+    /**
+     * Get the session variable name used to track member ID
+     *
+     * @return string
+     */
+    public function getSessionVariable()
+    {
+        return $this->sessionVariable;
+    }
+
+    /**
+     * Set the session variable name used to track member ID
+     *
+     * @param string $sessionVariable
+     * @return null
+     */
+    public function setSessionVariable($sessionVariable)
+    {
+        $this->sessionVariable = $sessionVariable;
+    }
+
+    /**
+     * @inherit
+     */
+    public function authenticateRequest(HTTPRequest $request)
+    {
+        // @todo couple the session to a request object
```

Comment:
That's going to be addressed by #6681. I don't think we'll tie session to request. I would leave this TODO out of this PR.